src/main/scala/beam/analysis/RoutingRequestAnalysis.scala

```diff
@@ -0,0 +1,50 @@
+package beam.analysis
+
+import beam.agentsim.events.ModeChoiceEvent
+import beam.analysis.plots.GraphAnalysis
+import beam.router.BeamRouter.RoutingRequest
+import beam.router.r5.RouteDumper.RoutingRequestEvent
+import beam.sim.config.BeamConfig
+import beam.utils.csv.CsvWriter
+import org.matsim.api.core.v01.events.Event
+import org.matsim.core.controler.events.IterationEndsEvent
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+case class CsvRow(personId: String, modeChoice: String, routingRequests: Vector[Int])
+
+class RoutingRequestAnalysis(beamConfig: BeamConfig) extends GraphAnalysis {
+
+  private val personRoutingIds: mutable.Map[String, Vector[Int]] = mutable
+    .Map[String, Vector[Int]]()
+    .withDefaultValue(Vector())
+  private val modeChoiceList = new ListBuffer[CsvRow]()
+
+  override def processStats(event: Event): Unit = {
+    event match {
+      case RoutingRequestEvent(RoutingRequest(_, _, _, _, personId, _, _, _, requestId)) if personId.isDefined =>
+        personRoutingIds(personId.get.toString) = personRoutingIds(personId.get.toString) :+ requestId
+      case modeChoiceEvent: ModeChoiceEvent =>
+        val personId = modeChoiceEvent.personId.toString
+        val routingIds = personRoutingIds.remove(personId).getOrElse(Vector.empty)
+        modeChoiceList.append(CsvRow(personId, modeChoiceEvent.mode, routingIds))
+      case _ =>
+    }
+  }
+
+  override def resetStats(): Unit = {
+    personRoutingIds.clear()
+    modeChoiceList.clear()
+  }
+
+  override def createGraph(event: IterationEndsEvent): Unit = {
+    val controller = event.getServices.getControlerIO
+    val filePath = controller.getIterationFilename(event.getIteration, "routingModeChoice.csv")
+    val csvWriter = new CsvWriter(filePath, IndexedSeq("PersonId", "ModeChoice", "RoutingRequestIds"))
```

Comment:
This should use Utils.using or try/finally to make sure the resource closes.

Reply:
Used Utils.using for file operations

Reply:
When you switched back to CsvWriter in the using you lost the using. Does CsvWriter somehow self-resolve or does it not implement Closeable? We should fix the latter so you can use Utils.using then.

Reply:
It is safer to create the instance as part of the `using` so it is not accidentally used after being disposed. I'll approve - but wanted to throw that out there.